### PR TITLE
Migrate to WorkManager

### LINKED ISSUE #3095: Migrate to WorkManager
>  In April 2019 we announced that Firebase Job Dispatcher would be deprecated today, April 7th 2020. For this reason we are going to close all open issues and archive the repository. We will also stop supporting FJD for new app installations (such as those targeting sdk versions greater than Android Q). Existing apps that rely on FJD should continue to function as usual.

https://github.com/googlearchive/firebase-jobdispatcher-android/issues/238

https://developer.android.com/topic/libraries/architecture/workmanager/migrating-fb

### PR DESCRIPTION
**Pull request by raghulvaikundam**
Migrate to WorkManager from Firebase JobDispatcher
Changed the JobService to ListenableWorker

Fixes #3095

### CODE REVIEW

**Comment by teolemon** [comment]
We need to check that there's no overlap with the work of @philippeauriach in #3072

**Comment by raghulvaikundam** [comment]
@teolemon sure. Do let me know if I need to make any changes based on @philippeauriach work

**Comment by VaiTon** [comment]
@teolemon just a little conflict with Utils.java. Nothing difficult.

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/jobs/SavedProductUploadWork.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+import androidx.annotation.NonNull;
+import androidx.work.ListenableWorker;
+import androidx.work.WorkerParameters;
+
+import com.google.common.util.concurrent.ListenableFuture;
+
+import openfoodfacts.github.scrachx.openfood.network.OpenFoodAPIClient;
+
+public class SavedProductUploadWork extends ListenableWorker {
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/jobs/SavedProductUploadWork.java:13]: > For most cases, we recommend using Worker, which offers a simple synchronous API that is executed on a pre-specified background thread.

From the docs.

Is there any reason for using `ListenableWorker`?

Comment (raghulvaikundam) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/jobs/SavedProductUploadWork.java:13]: > > For most cases, we recommend using Worker, which offers a simple synchronous API that is executed on a pre-specified background thread.
> 
> From the docs.
> 
> Is there any reason for using `ListenableWorker`?

Yes, we used `ListenableWorker` as we need to execute asynchronous rest API inside the worker.

**Comment by raghulvaikundam** [comment]
@VaiTon is it good for merging?

**Comment by VaiTon** [comment]
> 
> 
> @VaiTon is it good for merging?

I'm going to test it and then merge.

**Pull request merged**